### PR TITLE
update jwt claim based routing and add tests

### PR DESCRIPTION
**Pull request by yangminzhu**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [x] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/tasks/security/authentication/jwt-route/index.md
Comment (ericvn) [inline content/en/docs/tasks/security/authentication/jwt-route/index.md:79]: s/specifeid/specified/

Comment (yangminzhu) [reply content/en/docs/tasks/security/authentication/jwt-route/index.md]: thanks.

**Comment by yangminzhu** [comment]
note, the test failed because it was using the old proxy, rerunning the test to see if it now uses the latest.

**Comment by ericvn** [comment]
Note that the current tests reference the 1.12.0-beta.0 build. I do have a PR open to move to the beta.1, just waiting on its approval.

**Comment by yangminzhu** [comment]
@ericvn tests are now passing, ptal, thanks.

**Review by ericvn — APPROVED** [review]

**Pull request merged**
